packages/protocol/PaintsCache.ts

```diff
@@ -0,0 +1,56 @@
+import { createCache } from "suspense";
+
+import { recordingCapabilitiesCache } from "replay-next/src/suspense/BuildIdCache";
+import { findIndex } from "replay-next/src/utils/array";
+import { replayClient } from "shared/client/ReplayClientContext";
+import { TimeStampedPointWithPaintHash } from "shared/client/types";
+
+// This could be a streaming cache, but streaming APIs are more awkward to interop with
+// Since we wait for processing to complete before loading a recording, paints should always load very quickly
+export const PaintsCache = createCache<[], TimeStampedPointWithPaintHash[]>({
+  config: { immutable: true },
+  debugLabel: "PaintsCache",
+  getKey: () => ``, // Single entry cache
+  load: async ([]) => {
+    const recordingCapabilities = await recordingCapabilitiesCache.readAsync(replayClient);
+    if (recordingCapabilities.supportsRepaintingGraphics) {
+      return await replayClient.findPaints();
+    } else {
+      return [];
+    }
+  },
+});
+
+export function mostRecentPaint(time: number) {
+  const paints = PaintsCache.getValueIfCached() ?? [];
+
+  const index = findIndex(
+    paints,
+    { paintHash: "", point: "", time },
+    (a, b) => a.time - b.time,
+    false
+  );
+
+  // This is the nearest paint, but it may be before or after the time
```

Comment:
What binary search function finds the nearest entry that's before a value? To my knowledge, that's kind of an uncommon behavior. Most search functions find an exact match. (I don't see how `sortedIndexBy` does what you're suggesting.)

I think this is pretty simple though, and pretty clear about its intent. It will be replacing this method in a follow up PR:
https://github.com/replayio/devtools/blob/48cfadf6d704e4b275275d059a628f9f4c43afb0/packages/protocol/graphics.ts#L30-L47

Reply:
So why not use `binarySearch()` like this code does?

Reply:
The `binarySearch` method above is one in our codebase that has been written to return a `<=` match, rather than an exact match (or a closest match) which (I think) is not an obvious behavior, given its generic name.

This seems like a subjective preference.

Reply:
Scrap my previous reply. I'm context-switching between this PR and Domi chats and a coding task and I'm not doing a good job of it.

This feels like a subjective thing we're talking about (a nit) so I'd rather leave it as such, unless you feel strongly about it, in which case I will change it.

Reply:
Definitely just a nit.